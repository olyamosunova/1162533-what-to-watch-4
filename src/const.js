export const TabsName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const TABS = Object.values(TabsName);

export const GenreNames = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  KIDS_AND_FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`,
};

export const SCORE_NAME = {
  BAD: {
    name: `Bad`,
    min: 0,
    max: 3,
  },
  NORMAL: {
    name: `Normal`,
    min: 3,
    max: 5,
  },
  GOOD: {
    name: `Good`,
    min: 5,
    max: 8,
  },
  VERY_GOOD: {
    name: `Very good`,
    min: 8,
    max: 10,
  },
  AWESOME: {
    name: `Awesome`,
  },
};

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const CurrentPage = {
  MAIN: `MAIN`,
  DETAIL: `DETAIL`,
  LOGIN: `LOGIN`,
  PLAYER: `PLAYER`,
  ADD_REVIEW: `ADD_REVIEW`,
};

export const Review = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
};

export const TEXTAREA_BACKGROUNDCOLOR = `rgba(0, 0, 0, 0.15)`;

export const reviewSubmitButton = {
  post: `Post`,
  sending: `Sending...`,
};
