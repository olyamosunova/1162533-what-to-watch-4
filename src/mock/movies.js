const PromoMovie = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
};

const MoviesList = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
  },
  {
    id: 3,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `aviator.jpg`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 7,
    title: `Revenant`,
    poster: `revenant.jpg`,
  },
  {
    id: 8,
    title: `Johnny English`,
    poster: `johnny-english.jpg`,
  },
];

const MovieDescriptions = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
  RATING: 8.9,
  RATING_LEVEL: `Very good`,
  RATING_COUNT: 240,
  DESCRIPTION: [
    `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  ],
  DIRECTOR: `Wes Andreson`,
  STARRING: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
};

export {PromoMovie, MoviesList, MovieDescriptions};
