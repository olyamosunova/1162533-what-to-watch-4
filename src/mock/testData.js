const genres = [`Drama`, `Comedy`, `Western`];

const movie = {
  promoMovie: {
    id: 1,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
    cover: `macbeth.jpg`,
    genre: `Drama`,
    releaseDate: 2015,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  backgroundColor: ``,
  previewImage: ``,
  videoLink: ``,
  rating: 7.1,
  ratingLevel: `Very good`,
  ratingCount: 240,
  runTime: 120,
  description: ``,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
    `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
};

const Movies = [
  {
    promoMovie: {
      id: 1,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 2,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 3,
      title: `Macbeth`,
      poster: `macbeth.jpg`,
      cover: `macbeth.jpg`,
      genre: `Drama`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.1,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 4,
      title: `Aviator`,
      poster: `aviator.jpg`,
      cover: `aviator.jpg`,
      genre: `Drama`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 5,
      title: `We need to talk about Kevin`,
      poster: `we-need-to-talk-about-kevin.jpg`,
      cover: `we-need-to-talk-about-kevin.jpg`,
      genre: `Drama`,
      releaseDate: 2011,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.5,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 6,
      title: `What We Do in the Shadows`,
      poster: `what-we-do-in-the-shadows.jpg`,
      cover: `what-we-do-in-the-shadows.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 7,
      title: `Revenant`,
      poster: `revenant.jpg`,
      cover: `revenant.jpg`,
      genre: `Vestern`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.8,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 8,
      title: `Johnny English`,
      poster: `johnny-english.jpg`,
      cover: `johnny-english.jpg`,
      genre: `Comedy`,
      releaseDate: 2003,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.2,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 9,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 10,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
];

const reviews = [{
  id: 1,
  message: ``,
  rating: 8.9,
  author: ``,
  date: ``,
}];

const userData = {
  id: 0,
  email: `vesan@mail.ru`,
  name: `Lola`,
  avatarUrl: ``,
};

export {movie, Movies, reviews, genres, userData};
