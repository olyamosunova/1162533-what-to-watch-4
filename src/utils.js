export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenresList = (movies) => {
  return [`All genres`, ...new Set(movies.map(({promoMovie}) => promoMovie.genre))];
};
