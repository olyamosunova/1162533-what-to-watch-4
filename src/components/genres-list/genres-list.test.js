import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list.jsx";
import {GenreNames} from "./../../const.js";
import {Movies} from "../../mock/testData";
import {getGenresList} from "../../utils";

const genres = getGenresList(Movies);

it(`render GenresList`, () => {
  const tree = renderer
    .create(
        <GenresList
          activeGenre={GenreNames.ALL}
          onClick={()=>{}}
          genres={genres}
        />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
