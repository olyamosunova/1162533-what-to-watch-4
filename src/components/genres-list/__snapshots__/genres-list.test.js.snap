// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GenresList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Music
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Vestern
    </a>
  </li>
</ul>
`;
